<launch>
    
    <!-- select group_name: left_arm or right_arm -->
    <arg name="group_name" value="left_arm" />
    
    <!-- resolution in m (0.1, 0.05, 0.025) -->
    <arg name="r" value="0.1" />
	<!-- theta resolution -->
	<arg name="t" value="16" />
            
    <arg name="collision_checking" value="true" />
    
    <arg name="path_poly" value="$(find primitive_to_polygon)/polygons/tablew213h102.poly" />
    
	<!-- DO NOT CHANGE PARAMETERS BELOW -->

    <arg name="path_name" value="$(find inverse_capability_map_generator)/maps/inv_cap_table_pr2_$(arg group_name)_$(arg r)m_$(arg t)t.icpm" />
    <arg name="path_inv_obj" value="$(find inverse_capability_map_generator)/maps/inv_cap_object_pr2_$(arg group_name)_$(arg r)m_$(arg t)t.icpm" />
    
	<include file="$(find tidyup_pr2_moveit_config)/launch/planning_context.launch">
		<arg name="load_robot_description" value="true"/>
	</include>

    <node pkg="inverse_capability_map_generator" type="inverse_capability_table_generator"
    	name="inverse_capability_table_generator"
    	args="-p $(arg path_name) -o $(arg path_inv_obj)
    	-i $(arg path_poly)
    	-r $(arg r)"
    	output="screen" required="true">
    	<param name="collision_checking" type="bool" value="$(arg collision_checking)" />
    </node>
</launch>
